h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoder.java

```diff
@@ -60,11 +61,43 @@ public TargetEncoder (String[] columnNamesToEncode) {
       this(columnNamesToEncode, new BlendingParams(20, 10));
     }
 
-    public static class DataLeakageHandlingStrategy {
-        public static final byte LeaveOneOut  =  0;
-        public static final byte KFold  =  1;
-        public static final byte None  =  2;
+  public enum DataLeakageHandlingStrategy {
```

Comment:
Enums are more descriptive and easier to check against. So I've introduced it. The byte inside is only there for legacy reasons to keep the old API honor and intact. I believe the only external users are now Sparkling water people (cc @jakubhava ) and if they say we don't need the by-level API, I'll just remove it. There is a method to migrate from the old API to the new one, used on some places of the code. Could serve the same purpose.

Reply:
@Pscheidl I agree with you that enums are more convenient but for ultimate performance passing bytes(from clients or wherever) is more efficient. I was inspired by h2o-3 code base to use it this way.  But truth is somewhere in between of the usability and performance. @michalkurka Wanted to know if we have any considerations or conventions about it on the project?

Reply:
Sure. I am working with a Model and from user's perspective, I wanted strings with as few mapping layers in between as possible.